Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibsAPI.cs

```diff
@@ -0,0 +1,1006 @@
+        /// <param name="className">The name of the class document, as defined in the VBA project</param>
+        /// <returns>DocClassType indicating the type of the document class module, or DocType.Unrecognized</returns>
+        public static DocClassType DetermineDocumentClassType(TypeLibWrapper projectTypeLib, string className)
+        {
+            return DetermineDocumentClassType(projectTypeLib.TypeInfos.Get(className));
+        }
+
+        /// <summary>
+        /// Determines whether the specified document class is a known document class type (e.g. Excel._Workbook, Access._Form)
+        /// </summary>
+        /// <param name="project">Safe-com wrapper representing the VBA project</param>
+        /// <param name="className">The name of the class document, as defined in the VBA project</param>
+        /// <returns>DocClassType indicating the type of the document class module, or DocType.Unrecognized</returns>
+        public static DocClassType DetermineDocumentClassType(IVBProject project, string className)
+        {
+            using (var typeLib = TypeLibWrapper.FromVBProject(project))
+            {
+                return DetermineDocumentClassType(typeLib.TypeInfos.Get(className));
+            }
+        }
+
+        /// <summary>
+        /// Determines whether the specified document class is a known document class type (e.g. Excel._Workbook, Access._Form)
+        /// </summary>
+        /// <param name="project">Safe-com wrapper representing the VBA component</param>
+        /// <returns>DocClassType indicating the type of the document class module, or DocType.Unrecognized</returns>
+        public static DocClassType DetermineDocumentClassType(IVBComponent component)
+        {
+            return DetermineDocumentClassType(component.ParentProject, component.Name);
+        }
+
+        /// <summary>
+        /// Determines whether the specified document class is a known document class type (e.g. Excel._Workbook, Access._Form)
+        /// </summary>
+        /// <param name="classTypeInfo">Low-level ITypeInfo wrapper representing the VBA project</param>
+        /// <returns>DocClassType indicating the type of the document class module, or DocType.Unrecognized</returns>
+        public static DocClassType DetermineDocumentClassType(TypeInfoWrapper classTypeInfo)
+        {
+            return classTypeInfo.DetermineDocumentClassType();
+        }
+
+        /// <summary>
+        /// Determines whether the specified VBA class implements a specific interface
+        /// </summary>
+        /// <param name="ide">Safe-com wrapper representing the VBE</param>
+        /// <param name="projectName">VBA Project name, as declared in the VBE</param>
+        /// <param name="className">Document class name, as declared in the VBA project</param>
+        /// <param name="interfaceProgID">The interface name, preceeded by the library container name, e.g. "Excel._Worksheet"</param>
```

Comment:
Question - can this be used to inspect whether a VBA class implements a custom interface defined somewhere in same project using `Implements` statement?

Reply:
Implements statements do not appear in the type info unfortunately.  I will find a way around that at some point.